practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/controller/PractitionerController.java

```diff
@@ -71,6 +71,7 @@ public PractitionerController(EntityService entityService,  SecurityContextUtil
         return ResponseEntity.ok(entityService.findPractitionerById(id));
     }
 
+    @PreAuthorize("@authorisationService.authorise('assign-role', #roleAssignment.siteId)") //NOSONAR
```

Comment:
Are you sure you need a //NOSONAR annot?  I don't think sonar complains about preauth annotations?

Reply:
yes you have too, because sonar recognises it as a security issue, it ask to use role based access or other pre known validation. Since we are using custom authorisation it doesn't recognise it.